README.rst

```diff
@@ -655,6 +712,236 @@ any WSGI server, such as uWSGI or Gunicorn. For example:
+                raise falcon.HTTPError(falcon.HTTP_753,
+                                       title='Malformed JSON',
+                                       description='Could not decode the request body. The '
+                                       'JSON was incorrect or not encoded as '
+                                       'UTF-8.')
+
+        async def process_response(self, req, resp, resource, req_succeeded):
+            if not hasattr(resp.context, 'result'):
+                return
+
+            resp.body = json.dumps(resp.context.result)
+
+
+    def max_body(limit):
+
+        async def hook(req, resp, resource, params):
+            length = req.content_length
+            if length is not None and length > limit:
+                msg = ('The size of the request is too large. The body must not '
+                       'exceed ' + str(limit) + ' bytes in length.')
+
+                raise falcon.HTTPPayloadTooLarge(
+                    title='Request body is too large', description=msg)
+
+        return hook
+
+
+    class ThingsResource:
+
+        def __init__(self, db):
+            self.db = db
+            self.logger = logging.getLogger('thingsapp.' + __name__)
+
+        async def on_get(self, req, resp, user_id):
+            marker = req.get_param('marker') or ''
+            limit = req.get_param_as_int('limit') or 50
+
+            try:
+                result = await self.db.get_things(marker, limit)
+            except Exception as ex:
+                self.logger.error(ex)
+
+                description = ('Aliens have attacked our base! We will '
+                               'be back as soon as we fight them off. '
+                               'We appreciate your patience.')
+
+                raise falcon.HTTPServiceUnavailable(
+                    title='Service Outage',
+                    description=description,
+                    retry_after=30)
+
+            # NOTE: Normally you would use resp.media for this sort of thing;
+            # this example serves only to demonstrate how the context can be
+            # used to pass arbitrary values between middleware components,
+            # hooks, and resources.
+            resp.context.result = result
+
+            resp.set_header('Powered-By', 'Falcon')
+            resp.status = falcon.HTTP_200
+
+        @falcon.before(max_body(64 * 1024))
+        async def on_post(self, req, resp, user_id):
+            try:
+                doc = req.context.doc
+            except AttributeError:
+                raise falcon.HTTPBadRequest(
+                    title='Missing thing',
+                    description='A thing must be submitted in the request body.')
+
+            proper_thing = await self.db.add_thing(doc)
+
+            resp.status = falcon.HTTP_201
+            resp.location = '/%s/things/%s' % (user_id, proper_thing['id'])
+
+
+    # Configure your WSGI server to load "things.app" (app is a WSGI callable)
```

Comment:
Same as in some other place. An ASGI server.

Reply:
👍